### ISSUE TITLE
Hapi auth jwt typescript bug

### ISSUE THREAD

**Issue by vforv** [dwyl/hapi-auth-jwt2#310]
There is problem with types in last version.
when I do
tsc --traceResolution

get error:

`node_modules/hapi-auth-jwt2/lib/index.d.ts(1,66): error TS2307: Cannot find module '@hapi/hapi'.`

With old version: 8.1.0 works fine.

**Comment by xpolb01** [comment]
This affects previous versions too. Need a fix

**Comment by nelsonic** [comment]
PRs very much welcome.

**Comment by xpolb01** [comment]
I like the practice "one who broke it fixes it".

I have version 8.1.0 installed in my package.json. When I delete node_modules and do a fresh install it gives me this error on build:

```
node_modules/hapi-auth-jwt2/lib/index.d.ts:1:66 - error TS2307: Cannot find module '@hapi/hapi'.

1 import { Request, ResponseObject, Plugin, ResponseToolkit } from '@hapi/hapi';
```
Most people use CI/CD, in our case we use Jenkins, so fresh build happens every time as well as this error.

My PR would be rolling back this PR https://github.com/dwyl/hapi-auth-jwt2/pull/307/files until this bug is fixed because it is breaking previous versions

**Comment by jaulz** [comment]
@xpolb01 @vforv  are you still using `hapi` package instead of `@hapi/hapi`?

**Comment by xpolb01** [comment]
@jaulz Yes, but even if I replace it with @hapi/hapi this is still an issue. I'm using Glue so I think that is what is causing the problem.
In Glue plugins are defined like this:

```
import  * as jwtAuth from 'hapi-auth-jwt2';
const jwtPlugin: Glue.Plugin = { plugin: jwtAuth, options: {} };

manifest
const manifest: Glue.Manifest = {
  server: {
    port: process.env.PORT || 5567,
    tls: tlsConfig,
  },
  register: {
    plugins: [jwtPlugin, apiPlugin, hapiBoomDecoratorsPlugin, sequelizePlugin],
  },
};
```

By the way, this only works if I user require instead of import. Otherwise it gives me an error:
Type 'Plugin<Re```
gisterOptions>' is not assignable to type 'string | { register: string; options?: any; }'.
  Type 'PluginBase<RegisterOptions> & PluginNameVersion' is not assignable to type 'string | { register: string; options?: any; }'.
    Type 'PluginBase<RegisterOptions> & PluginNameVersion' is not assignable to type '{ register: string; options?: any; }'.
      Types of property 'register' are incompatible.
        Type '(server: Server, options: RegisterOptions) => void | Promise<void>' is not assignable to type 'string'.ts(2322)
```

**Comment by jaulz** [comment]
I think it should rather be `import jwtAuth from 'hapi-auth-jwt2';`.

**Comment by xpolb01** [comment]
In that case, I will get this error:

```

Module '"/Users/bogdan/Desktop/Work/betterpt-k8s/notification-service-k8s/node_modules/hapi-auth-jwt2/lib/index"' can only be default-imported using the 'esModuleInterop' flagts(1259)
index.d.ts(131, 1): This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' 
```

**Comment by jaulz** [comment]
And what about this?

```
import  * as jwtAuth from 'hapi-auth-jwt2';
const jwtPlugin: Glue.Plugin = { plugin: jwtAuth.plugin, options: {} };
```

By the way, which hapi types are you using?

**Comment by xpolb01** [comment]
I'm using **@types/hapi__hapi**.
That won't work either cause, it says that it has no property **plugin** on it.

Error:
```
Property 'plugin' does not exist on type 'Plugin<RegisterOptions>'.
  Property 'plugin' does not exist on type 'PluginBase<RegisterOptions> & PluginNameVersion'.ts(233
```

**Comment by jaulz** [comment]
Right, that was my fault. Which glue types are you using?

**Comment by xpolb01** [comment]
I use "@types/glue": "^5.0.3",

**Comment by jaulz** [comment]
Then should you probably also use `@hapi/glue`.

**Comment by xpolb01** [comment]
Yep, that works.
But I think what I was trying to point to, that this forced me to update hapi, and the whole thing, even though I was using version 8.1. Service was working fine on development and then it suddenly started failing after restart. Somehow 8.6 affected previous versions which are in TS.

**Comment by nelsonic** [comment]
@xpolb01 / @jaulz still an issue?
